crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -130,8 +130,14 @@ def end_step(success = true, errors = {})
       end
     end
 
+    # Check if given step is running.
+    # Without argument, check if any step is running
     def running?(step_name = nil)
-      step = progress[:steps][step_name || current_step]
+      if step_name.nil?
+        return progress[:steps].select { |_, s| s[:status] == :running }.any?
```

Comment:
only a nit, but I find it good to use decent naming, even if they are gonna be discarded, when doing one liners as to be more clear on a first look. This had me looking at it for a good 30 seconds trying to understand what that strange `|_` was until I got that its a var name that its not gonna be used.